query-node/mappings/src/content/utils.ts

```diff
@@ -0,0 +1,484 @@
+    if (metaAsObject.avatarPhoto !== undefined) {
+      const asset = await extractAsset({
+        assetIndex: metaAsObject.avatarPhoto,
+        assets: parameters.assets,
+        db: parameters.db,
+        blockNumber: parameters.blockNumber,
+        contentOwner: parameters.contentOwner,
+      })
+      integrateAsset('avatarPhoto', result, asset) // changes `result` inline!
+      delete metaAsObject.avatarPhoto
+    }
+
+    // prepare language if needed
+    if (metaAsObject.language) {
+      result.language = await prepareLanguage(metaAsObject.language, parameters.db, parameters.blockNumber)
+    }
+
+    return result as Partial<T>
+  }
+
+  // process video
+  if (type instanceof Video) {
+    const meta = VideoMetadata.deserializeBinary(metaU8a)
+    const metaAsObject = meta.toObject()
+    const result = metaAsObject as any as Partial<Video>
+
+    // prepare video category if needed
+    if (metaAsObject.category !== undefined) {
+      result.category = await prepareVideoCategory(metaAsObject.category, parameters.db)
+    }
+
+    // prepare media meta information if needed
+    if (metaAsObject.mediaType) {
+      // prepare video file size if poosible
+      const videoSize = await extractVideoSize(parameters.assets, metaAsObject.video)
+
+      result.mediaMetadata = await prepareVideoMetadata(metaAsObject, videoSize)
+      delete metaAsObject.mediaType
+    }
+
+    // prepare license if needed
+    if (metaAsObject.license) {
+      result.license = await prepareLicense(metaAsObject.license)
+    }
+
+    // prepare thumbnail photo asset if needed
+    if (metaAsObject.thumbnailPhoto !== undefined) {
+      const asset = await extractAsset({
+        assetIndex: metaAsObject.thumbnailPhoto,
+        assets: parameters.assets,
+        db: parameters.db,
+        blockNumber: parameters.blockNumber,
+        contentOwner: parameters.contentOwner,
+      })
+      integrateAsset('thumbnail', result, asset) // changes `result` inline!
+      delete metaAsObject.thumbnailPhoto
+    }
+
+    // prepare video asset if needed
+    if (metaAsObject.video !== undefined) {
+      const asset = await extractAsset({
+        assetIndex: metaAsObject.video,
+        assets: parameters.assets,
+        db: parameters.db,
+        blockNumber: parameters.blockNumber,
+        contentOwner: parameters.contentOwner,
+      })
+      integrateAsset('media', result, asset) // changes `result` inline!
+      delete metaAsObject.video
+    }
+
+    // prepare language if needed
+    if (metaAsObject.language) {
+      result.language = await prepareLanguage(metaAsObject.language, parameters.db, parameters.blockNumber)
+    }
+
+    // prepare information about media published somewhere else before Joystream if needed.
+    if (metaAsObject.publishedBeforeJoystream) {
+      // this will change the `channel`!
+      handlePublishedBeforeJoystream(result, metaAsObject.publishedBeforeJoystream.date)
```

Comment:
we should rely on the `publishedBeforeJoystream.isPublished` boolean to determine if the caller intended to set the date instead of relying on wether the date field is a string or not.

Reply:
Check for `isPublished` added.